src/docs/files-provider-deprecation.rst

```diff
@@ -0,0 +1,60 @@
+Files Provider Removal
+######################
+
+The SSSD team has announced the `removal of the files provider <https://pagure.io/fesco/issue/3107>`_ feature in recent versions of SSSD in Fedora/CentOS Stream/RHEL. This document describes how to switch to using the SSSD 'proxy' provider to replace files provider functionality.
+
+Practically, there are only two use cases that currently justify usage of the files provider feature:
+
+* smart card authentication of local users
+* session recording for local users
+
+For both cases the proxy provider is a viable substitute. Otherwise, files provider functionality does not provide any benefit and should remain disabled.
+
+Proxy Provider Configuration
+****************************
+
+The SSSD proxy provider is just a relay, an intermediary configuration. SSSD connects to its proxy service, and then that proxy loads the specified libraries. This allows SSSD to use some resources that it otherwise would not be able to use.
+
+First, make sure to install the ``sssd-proxy`` package.
+
+.. code-block:: console
+
+    $ dnf install sssd-proxy
+
+A sample proxy provider configuration to load the nss `files` module is below:
+
+.. code-block:: console
+
+    [sssd]
+    services=nss, pam
+    domains=nssfiles
+
+    [domain/nssfiles]
+    id_provider=proxy
+    proxy_lib_name=files
+    proxy_pam_target=sssd-shadowutils
+
+The ``proxy_lib_name`` option specifies which existing NSS library to proxy identity requests through.
+
+The ``proxy_pam_target`` specifies the target to which PAM must proxy as an authentication provider. This PAM target is a file containing PAM stack information in the default PAM directory, ``/etc/pam.d/``. The following ``sssd-shadowutils`` file exists on RHEL-based Linux systems.
+
```

Comment:
Maybe better reference available upstream https://github.com/SSSD/sssd/blob/master/src/examples/sssd-shadowutils ?

Reply:
Upstream sssd.spec packages this file:
https://github.com/SSSD/sssd/blob/998503210b2644dda35091ce87531d3ee31a94b4/contrib/sssd.spec.in#L808
And Fedora ships it.

Reply:
Updated.